lms/djangoapps/courseware/access.py

```diff
@@ -370,17 +396,24 @@ def _can_load_course_overview(user, course_overview):
         The user doesn't have to be enrolled in the course in order to have load
         load access.
     """
-    return (
-        not course_overview.visible_to_staff_only
-        and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
-    ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+    if _has_staff_access_to_descriptor(user, course_overview, course_overview.id):
```

Comment:
this section can't be combined completely because for this part 
`if course_overview.visible_to_staff_only:
        return VisibilityError()`
it has to return a new object and not the one in the boolean expression